django/apps/aggregated/management/commands/update_aggregated_data.py

```diff
@@ -23,13 +23,15 @@
     )
     (
         -- Retrieve used tasks
+        -- task_id is not the primary key. project_id-group_id-task_id is
         WITH used_tasks as (
             SELECT
-              project_id, group_id, task_id
-            FROM results R
-              INNER JOIN tasks T USING (project_id, group_id, task_id)
+              MS.project_id, MS.group_id, MSR.task_id
+            FROM mapping_sessions_results MSR
```

Comment:
Hey @thenav56 ,
I would try to not use the `mapping_sessions_results` table here, but only use the `mapping_sessions` if possible.

you should be able to join the tasks for each mapping session using the `project_id` and `group_id` attributes.

Reply:
Hi @Hagellach37 
I would also like to skip touching mapping_sessions_results, but we are going through the task for the area.
If you think we can group (aggregate sum) all task areas by project_id and group_id and then use that for project_id and group_id in the session then that sounds good.

Could you confirm that? I will change the logic according to that.

Reply:
yes it should be enough that we always group the task area per group. We do not need to have the area for each individual task, but it is enough to have it per mapping session

Reply:
Sounds good. Thanks.

I have pushed recommended changes.